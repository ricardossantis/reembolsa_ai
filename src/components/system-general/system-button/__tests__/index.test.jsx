import React from "react";
import renderer from "react-test-renderer";
import SystemButton from "../index";

it("Emulating cancel", () => {
  const tree = renderer
    .create(
      <SystemButton
        cancel={() => {
          "Operação cancelada";
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Emulating confirm", () => {
  const tree = renderer
    .create(
      <SystemButton
        confirm={() => {
          "Operação confirmada";
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
