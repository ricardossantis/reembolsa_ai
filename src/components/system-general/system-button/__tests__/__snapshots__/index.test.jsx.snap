// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emulating cancel 1`] = `
<div
  className="sc-bdnylx eWgwUX"
>
  <span
    aria-label="close-circle"
    className="anticon anticon-close-circle sc-gtssRu dJEhrD"
    onClick={[Function]}
    role="img"
    tabIndex={-1}
  >
    <svg
      aria-hidden="true"
      className=""
      data-icon="close-circle"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
      />
    </svg>
  </span>
  <span
    aria-label="check-circle"
    className="anticon anticon-check-circle sc-dlnjPT jZzMWT"
    role="img"
  >
    <svg
      aria-hidden="true"
      className=""
      data-icon="check-circle"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
      />
    </svg>
  </span>
</div>
`;

exports[`Emulating confirm 1`] = `
<div
  className="sc-bdnylx eWgwUX"
>
  <span
    aria-label="close-circle"
    className="anticon anticon-close-circle sc-gtssRu dJEhrD"
    role="img"
  >
    <svg
      aria-hidden="true"
      className=""
      data-icon="close-circle"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
      />
    </svg>
  </span>
  <span
    aria-label="check-circle"
    className="anticon anticon-check-circle sc-dlnjPT jZzMWT"
    onClick={[Function]}
    role="img"
    tabIndex={-1}
  >
    <svg
      aria-hidden="true"
      className=""
      data-icon="check-circle"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
      />
    </svg>
  </span>
</div>
`;
